gms/gossiper.cc

```diff
@@ -421,6 +425,56 @@ future<> gossiper::handle_echo_msg() {
     return make_ready_future<>();
 }
 
+// Must run inside a seastar thread
+uint64_t gossiper::get_gossip_status_token_hash() const {
+    auto nodes = boost::copy_range<std::set<gms::inet_address>>(endpoint_state_map | boost::adaptors::map_keys);
+    if (nodes.empty()) {
+        return 0;
+    }
+    xx_hasher h;
+    bytes_ostream out;
+    for (const auto& node : nodes) {
+        thread::maybe_yield();
+        out.clear();
+        auto state = get_application_state_ptr(node, application_state::TOKENS);
+        auto tokens = state ? state->value : "";
+        ser::serialize(out, tokens);
+        ser::serialize(out, get_gossip_status(node));
+        ser::serialize(out, node);
+        feed_hash(h, out.linearize());
+    }
+    return h.finalize_uint64();
+}
+
+static future<gossip_query_token_status_response>
+get_gossip_query_token_status_response() {
+    using response = gossip_query_token_status_response;
+    return _the_gossiper.map([] (gossiper& g) -> future<gossip_query_token_status_response> {
+        return seastar::async([&g] {
+            auto& tm = service::get_local_storage_service().get_token_metadata();
+            gossip_query_token_status_response resp = {{
```

Comment:
`response`, why not use the typedef introduced at the beginning

Reply:
Yes, we can. Fixed.